ProteGO/Common/DangerStatusManager/DangerStatusManager.swift

```diff
@@ -2,29 +2,45 @@ import Foundation
 import SwiftTweaks
 import RxSwift
 import RxCocoa
+import Valet
 
 final class DangerStatusManager: DangerStatusManagerType {
+
     var currentStatus: BehaviorRelay<DangerStatus>
 
     private var tweakBindings = Set<TweakBindingIdentifier>()
 
-    init() {
-        self.currentStatus = BehaviorRelay<DangerStatus>(value: .yellow)
+    private let gcpClient: GcpClientType
 
-        if DebugMenu.assign(DebugMenu.forceDangerStatus) {
-            if let status = DangerStatus.init(rawValue: DebugMenu.assign(DebugMenu.forceDangerStatusValue).value) {
-                self.currentStatus.accept(status)
-            }
+    private let valet: Valet
+
+    private let disposeBag = DisposeBag()
+
+    init(gcpClient: GcpClientType, valet: Valet) {
+        self.gcpClient = gcpClient
+        self.valet = valet
 
-            tweakBindings.insert(DebugMenu.bind(DebugMenu.forceDangerStatusValue) { value in
-                if let status = DangerStatus.init(rawValue: value.value) {
-                    self.currentStatus.accept(status)
-                }
-            })
-        }
+        let initialDangerStatus = valet.string(forKey: Constants.KeychainKeys.currentDangerStatus)
+            .flatMap(DangerStatus.init(rawValue:)) ?? .yellow
+
+        self.currentStatus = BehaviorRelay<DangerStatus>(value: initialDangerStatus)
     }
 
-    deinit {
-        self.tweakBindings.forEach(DebugMenu.unbind)
+    func updateCurrentDangerStatus() {
+        return self.gcpClient.getStatus().subscribe(onSuccess: { [weak self] result in
+            guard let self = self else {
+                logger.error("Instance deallocated file: \(#file), line: \(#line)")
+                return
+            }
+
+            switch result {
+            case .success(let result):
+                self.currentStatus.accept(result.status)
+                self.valet.set(string: result.status.rawValue, forKey: Constants.KeychainKeys.currentDangerStatus)
```

Comment:
Czy rzeczywiście tak powinien zachować się komponent? Jeżeli self przestanie istnieć do zachowanie zakończenia subskrybcji przestanie zapisywać tą wartość. może użycie capture listy na `valet` miałoby większy sens:

```
 return self.gcpClient.getStatus().subscribe(onSuccess: { [weak self, valet] result in
             guard let self = self else {
                 logger.error("Instance deallocated file: \(#file), line: \(#line)")
                 return
             }

             switch result {
             case .success(let result):
                 self.currentStatus.accept(result.status)
                 valet.set(string: result.status.rawValue, forKey: Constants.KeychainKeys.currentDangerStatus)
```

weak self będzie tylko pomijało stan, który aktualnie może zostać pominięty bo i tak nie zostanie obsłużony gdyby self przestało istnieć ale nadal `valet` ustawi wartość i zapytanie do serwera nie zostanie pominięte.

Reply:
Ok, rozumiem, że chciałbyś tutaj obronić się jakiś sposób przed hipotetyczną sytuacją gdzie self przestał istnieć, ale valet nadal istnieje?

Abstrachując od tego jak bardzo jest to nieprawdopodobne to powodowałoby to zachwiałoby to spójność stanu tego obiektu - w valecie mielibyśmy dobry stan, ale currentStatus by miał błędny.

W takim przypadku wolę już w pełni anulowac przypisywanie stanu i wylogowac info o błędzie (tak jak dzieje się teraz)

Reply:
👍